File: src/main/java/org/amfibot/discord/api/DiscordAPI.java
package org.amfibot.discord.api;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@SpringBootApplication
@RestController
public class DiscordAPI {

	public static void main(String[] args) {
		SpringApplication.run(DiscordAPI.class, args);
	}

	@GetMapping("/")
	public String root() {
		return "Hello, discord api!";
	}

}

File: src/main/kotlin/org/amfibot/discord/api/DiscordAPI.kt
package org.amfibot.discord.api

import org.apache.catalina.webresources.TomcatURLStreamHandlerFactory
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.security.core.Authentication
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@SpringBootApplication
@RestController
class DiscordAPI {
    @GetMapping("/")
    fun root(): String {
        return "Hello, discord api!"
    }

    @GetMapping("/user")
    fun getUser(authentication: Authentication): Authentication {
        return authentication
    }

}

fun main(args: Array<String>) {
    // Disables TomcatURLStreamHandlerFactory
    // to bypass the factory is already defined exception
    // when auto reloading the application with spring DevTools remote
    TomcatURLStreamHandlerFactory.disable()

    runApplication<DiscordAPI>(*args)
}